r/R/utilities_markers.R

```diff
@@ -17,9 +19,11 @@
 #' @return data.frame of top marker genes
 #' @export
 #'
-getTopMarkerGenes <- function(nFeatures, data, cellSets, aucMin = 0.3, pctInMin = 20, pctOutMax = 70) {
+getTopMarkerGenes <- function(nFeatures, data, cellSets, cellSetsIds = c(), aucMin = 0.3, pctInMin = 20, pctOutMax = 70) {
```

Comment:
cellSetIds! let's not get a second cells_id if we can avoid it. 🙏 

```suggestion
getTopMarkerGenes <- function(nFeatures, data, cellSets, cellSetIds = c(), aucMin = 0.3, pctInMin = 20, pctOutMax = 70) {
```

Reply:
where are they used? why is it empty?

Reply:
ah, it's related to the memoisation right?

Reply:
they aren't used by the function directly, they are used by the memoised function. If you look at the part where we create `memoisedGetTopMarkerGenes`, we ignore `cellSets` becasue we don't care if the user wanted to change the name of a cell set to determine the top marker genes, we only care about the cell ids. This is why we extract the cellIds and pass them apart from the rest.

> cellSetIds! let's not get a second cells_id if we can avoid it. 
I don't agree in this case, the object we are extracting the ids from is called `cellSets`, and we want the ids within that, so that's why it's called `cellSetsIds`

Reply:
I assumed it was memoization stuff. I don't know how it works, so can't really comment on this. Let me look a bit at how this function works.

Regarding the names. I prefer not writing incorrectly when there's no need to do so. In any case, you're passing a vector of cell ids (and not ids of cell sets), why not use `cell_ids`?

Reply:
Resolved in a slack call